src/sql/workbench/contrib/notebook/browser/notebook.component.ts

```diff
@@ -104,18 +106,23 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 		@Inject(ITextFileService) private textFileService: ITextFileService,
 		@Inject(ILogService) private readonly logService: ILogService,
 		@Inject(ICommandService) private commandService: ICommandService,
-		@Inject(IAdsTelemetryService) private adstelemetryService: IAdsTelemetryService
+		@Inject(IAdsTelemetryService) private adstelemetryService: IAdsTelemetryService,
+		@Inject(IConfigurationService) private _configurationService: IConfigurationService
 	) {
 		super();
 		this.updateProfile();
 		this.isLoading = true;
+		this._register(this._configurationService.onDidChangeConfiguration(e => {
+			this.previewFeaturesEnabled = this._configurationService.getValue('workbench.enablePreviewFeatures');
+		}));
 	}
 
 	private updateProfile(): void {
 		this.profile = this.notebookParams ? this.notebookParams.profile : undefined;
 	}
 
 	ngOnInit() {
+		this.previewFeaturesEnabled = this._configurationService.getValue('workbench.enablePreviewFeatures');
```

Comment:
You can just query the configuration service directly in initToolbar - no need to cache it like this

Reply:
I noticed that if I uncheck the preview features flag, I still have to reload the app to see the changes take effect. When I did this for the markdown toolbar, I set the value of this.previewFeaturesEnabled in both the constructor and ngOnInit. Is there a more efficient way of doing this?

Reply:
Well - so the value will be updated immediately. But you'd need to hook into getting notified when the value is updated if you want to update already open editors. That's what the onDidChangeConfiguration event is for : https://github.com/microsoft/azuredatastudio/blob/master/src/sql/workbench/contrib/query/browser/queryEditor.ts#L189

But if you open a new editor it should reading in the current values when it gets created.

That being said - I would generally say that for UI like this just checking it when the UI is created is fine. Trying to make it able to handle the configuration changes can sometimes be tricky so just having the setting be applied going forward has generally been our approach.